versions.tf

```diff
@@ -1,3 +1,9 @@
 terraform {
   required_version = ">= 0.12"
+  required_providers {
```

Comment:
How come **http** is a required_provider now? I understand before it was "included" and that's why on `main.tf` we could do `data.http...`.
Also I think it makes sense to force the 2.2.0 or greater version since it deprecates the upgraded attribute, but why put a ceiling with `< 4.0.0` if the latest release is the v3.1.0 (released yesterday)?

Reply:
Until now it was an implicit requirement, due to the presence of an http data block. But in order to constrain the upper version it must be explicitly listed.

The upper version limit is for safety. Because hashicorp/http abides by semantic versioning and the 3.0.0 release already broke this module, it should have been constrained long ago. Until a 4.0.0 is released we can't yet know if the module will compatible so it is safer to only increase the upper limit AFTER those releases exist and are known to work.